src/interpreter/scheme/base.rs

```diff
@@ -152,6 +152,26 @@ pub fn base_library<'a, R: RealNumberInternalTrait, E: IEnvironment<R>>(
         }
     }
 
+    fn floor<R: RealNumberInternalTrait, E: IEnvironment<R>>(
+        arguments: impl IntoIterator<Item = Value<R, E>>,
+    ) -> Result<Value<R, E>> {
+        match arguments.into_iter().next() {
+            Some(Value::Number(number)) => Ok(Value::Number(match number {
+                Number::Integer(num) => Number::Integer(num),
+                Number::Real(num) => Number::Integer(num.floor().to_i32().unwrap()),
```

Comment:
I think floor of Real should be Real, like `floorf()` in C. Otherwise it may be overflow.